torch/nn/modules/activation.py

```diff
@@ -1062,7 +1062,15 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
         if not is_batched:
             why_not_fast_path = f"input not batched; expected query.dim() of 3 but got {query.dim()}"
         elif query is not key or key is not value:
+            # When lifting this restriction, don't forget to either
+            # enforce that the dtypes all match or test cases where
+            # they don't!
             why_not_fast_path = "non-self attention was used (query, key, and value are not the same Tensor)"
+        elif query.dtype != self.in_proj_bias.dtype:
+            why_not_fast_path = f"dtypes of query ({query.dtype}) and self.in_proj_bias ({self.in_proj_bias.dtype}) don't match"
+        elif query.dtype != self.in_proj_weight.dtype:
```

Comment:
Hey @swolchok, just commenting that this broke our test in MaskedTensor here: https://github.com/pytorch/maskedtensor/blob/main/test/test_basics.py#L45.

Of note:
```
attn_nn = torch.nn.MultiheadAttention(1, 1, bias=False)
attn_nn(
    x, x, x, key_padding_mask=key_padding_mask, attn_mask=attn_mask
)
```
fails at this check because it seems that when `bias=False`, then `self.in_proj_bias = None`, and `None` does not have a `dtype`

We could fix it on our end, but I figure that's something you might want to check for as well